### PR TITLE
Hardening 5

### PR DESCRIPTION
**Pull request by gregaubert**

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/settings/store/values/reducer.js [hunk 1]
```diff
@@ -18,14 +18,35 @@
...
+  const key = action.componentKey;
+  if (action.type === RECEIVE_VALUES) {
+    const settingsByKey = keyBy(action.settings, 'key');
+    return { ...state, [key]: { ...(state[key] || {}), ...settingsByKey } };
+  }
+
+  return state;
+};
+
+const globalSettings = (state /*: SettingsState */ = {}, action /*: Object */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/settings/store/values/reducer.js]: MAJOR Move parameter "state" after parameters without default value. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS1788)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: server/sonar-web/src/main/js/api/web-api.ts [hunk 2]
```diff
@@ -50,6 +57,12 @@ export interface Domain {
   deprecated: boolean;
   internal: boolean;
   path: string;
+  since?: string;
+}
+
+export interface Exemple {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/api/web-api.ts]: Example

File: server/sonar-web/src/main/js/apps/web-api/components/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            value={this.state.query}
+            placeholder={translate('search_verb')}
+            onChange={this.handleSearch}
+          />
+        </div>
+
+        <div className="big-spacer-top">
+          <Checkbox checked={showInternal} onCheck={onToggleInternal}>
+            <span className="little-spacer-left">
+              {translate('api_documentation.show_deprecated')}
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/web-api/components/Search.tsx]: The label is wrong, it should be "show internal". And another one is wrong too.

File: server/sonar-web/src/main/js/apps/web-api/utils.tsx [hunk 4]
```diff
@@ -17,15 +17,14 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-//@flow
-/*:: import type { Domain, Action } from '../../api/web-api'; */
+import { Domain, Action } from '../../api/web-api';
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/web-api/utils.tsx]: this file can be `.ts`

File: server/sonar-web/src/main/js/apps/web-api/components/WebApiApp.tsx [hunk 5]
```diff
@@ -85,13 +88,12 @@ export default class WebApiApp extends React.PureComponent {
...
 
-  scrollToElement(id /*: string */) {
+  scrollToElement(id: string) {
     const element = document.getElementById(id);
 
     if (element) {
       const rect = element.getBoundingClientRect();
       const top = rect.top + window.pageYOffset - 20;
-
       window.scrollTo(0, top);
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/web-api/components/WebApiApp.tsx]: actually `helpers/scrolling.ts` can be used here

File: server/sonar-web/src/main/js/apps/web-api/components/__tests__/Domain-test.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  internal: false
+};
+const DEFAULT_PROPS = {
+  domain: DOMAIN,
+  showDeprecated: false,
+  showInternal: false,
+  searchQuery: ''
+};
+
+it('should render deprecated actions', () => {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/web-api/components/__tests__/Domain-test.tsx]: well, could be nice to add more tests to cover all components 😉

Comment (gregaubert) [reply server/sonar-web/src/main/js/apps/web-api/components/__tests__/Domain-test.tsx]: alright let's do this!

**Comment by gregaubert** [comment]
@stas-vilchik tests added! and fixes done!

**Pull request merged**